### PR TITLE
Changes required to set minimum Python version to 3.7

### PR DESCRIPTION
**Pull request by LucyJimenez**
xref #1022 

Update the required Python version on `setup.py` and some parts of the documentation.

@datapythonista do you think we need to update the examples on the exclude and include part? https://asv.readthedocs.io/en/stable/asv.conf.json.html

### CODE REVIEW

File: docs/source/installing.rst [hunk 1]
```diff
@@ -2,7 +2,7 @@ Installing airspeed velocity
 ============================
 
 **airspeed velocity** is known to work on Linux, Mac OS-X, and Windows.
-It is known to work with Python 2.7, 3.4, 3.5, and 3.6.
+It is known to work with Python 3.7.
```
Comment (datapythonista) [inline docs/source/installing.rst:5]: do you mind saying `with Python 3.7 and higher.` please?

File: docs/source/using.rst [hunk 2]
```diff
@@ -214,11 +214,11 @@ installed to support other environment tools.  The
 tool used to create environments.
 
 When using ``virtualenv``, ``asv`` does not build Python interpreters
-for you, but it expects to find each of the Python versions specified
+for you, but it expects to find the Python versions specified
 in the ``asv.conf.json`` file available on the ``PATH``.  For example,
 if the ``asv.conf.json`` file has::
 
-  "pythons": ["2.7", "3.6"]
+  "pythons": ["3.7"]
```
Comment (datapythonista) [inline docs/source/using.rst:221]: I think the example is clearer if you use 2 versions, maybe 3.7 and 3.10.

**Review by datapythonista — COMMENTED** [review]
Thanks for the changes @LucyJimenez 

I think updating the versions in the asv.conf.json page makes sense. IMO it looks better not using versions so much in the past, and maybe a bit misleading to show unsupported versions. But up to you.

**Comment by LucyJimenez** [comment]
> I think updating the versions in the asv.conf.json page makes sense. IMO it looks better not using versions so much in the past, and maybe a bit misleading to show unsupported versions. But up to you.

Perfect, I'll do it!

**Review by datapythonista — APPROVED** [review]
Thanks for taking care of this @LucyJimenez, looks great!

**Pull request merged**